engine/api/warning/warning_variable.go

```diff
@@ -7,9 +7,74 @@ import (
 	"github.com/ovh/cds/engine/api/environment"
 	"github.com/ovh/cds/engine/api/pipeline"
 	"github.com/ovh/cds/engine/api/workflow"
+	"github.com/ovh/cds/sdk"
 	"github.com/ovh/cds/sdk/log"
+	"regexp"
+	"strings"
+	"time"
 )
 
+var projVarRegexp *regexp.Regexp
+
+func Init() error {
+	var err error
+	projVarRegexp, err = regexp.Compile("cds\\.proj\\.[a-zA-Z0-9\\-_]+")
```

Comment:
should use raw string (`...`) with regexp.Compile to avoid having to escape twice